### PR TITLE
Allow customising CONFIGDIR

### PR DESCRIPTION
**Pull request by mplatings**
When building multiple branches of my project simultaneously, PyInstaller simultaneously tries to modify the same files. By specifying different config directories, I can avoid this problem.

### CODE REVIEW

**Comment by htgoebel** [comment]
Thanks, Using an environment variable and not a command-line options for this seams to be appropriate to me, because this should be rarely used. 

I'm curious, which files are modified simultaneously. The config-dir basically holds the cache.

To complete the pull-request, please document this in all of the man-pages and the manual.

**Comment by mplatings** [comment]
Thanks Hartmut, will this do?

> I'm curious, which files are modified simultaneously.
I should have specified "multiple branches on multiple platforms". Hopefully what I added to the manual will make it clear.

**Comment by htgoebel** [comment]
@mplatings Thanks for updating the manuals :-) When reading 9e57cf3, I understood why this is necessary. Thanks for this explanation.
I wonder if we couldn't fix the root cause, e.g by adding the platform to the name of the cache-dir. Would this work? What do you think?

**Comment by htgoebel** [comment]
@mplatings Whould #1269 solve your case, too?

**Comment by mplatings** [comment]
@htgoebel We could have the same architecture on Linux & Mac so I don't think #1269 would be sufficient. However, yes I think your suggestion of adding the platform name (and the architecture too) would do the trick, and would be a more elegant solution :)

**Comment by codewarrior0** [comment]
There might still be some merit to CONFIGDIR. Sharing a home directory between VMs is kind of unusual, but I can imagine a use-case of building two different branches of a codebase on the same machine using the same python, cpu, arch, etc...

File: PyInstaller/__init__.py [hunk 1]
```diff
@@ -78,7 +78,9 @@
             pass
 
 
-if is_win:
+if os.environ.get('PYINSTALLERCONFIGDIR'):
```
Comment (matysek) [inline PyInstaller/__init__.py]: @mplatings I could you use underscores in this variable name like: PYINSTALLER_CONFIG_DIR

**Comment by matysek** [comment]
@mplatings Thanks also for updating the manual. We will need this option for running tests in parallel.

**Comment by mplatings** [comment]
@matysek Done.

**Pull request merged**